app/controllers/posts_controller.rb

```diff
@@ -1,9 +1,30 @@
 class PostsController < ApplicationController
   def index
-    @user = User.find(params[:user_id])
+    @user = current_user
+    @posts = @user.posts
   end
 
   def show
     @post = Post.find(params[:id])
   end
+
+  def new
+    @new_post = Post.new
+  end
+
+  def create
+    puts params
+    user = current_user
+    new_post = Post.new(post_params)
+    new_post.author = user
+    if new_post.save
+      redirect_to user_posts_url(id: user.id)
+    else
+      redirect_to new_post_url(user_id: user.id)
+    end
+  end
+
+  def post_params
+    params.require(:post).permit(:Title, :Text)
+  end
```

Comment:
- [ ] when I tried to create a post I got the below error please make sure to fix it in order for we can test other parts like comments and like 
<img width="997" alt="Screenshot 2023-02-03 at 17 37 06" src="https://user-images.githubusercontent.com/61790587/216657724-6227aa50-69c6-4ba2-98bb-d088a96c3128.png">